homeassistant/util/network.py

```diff
@@ -39,3 +41,24 @@ def is_link_local(address: Union[IPv4Address, IPv6Address]) -> bool:
 def is_local(address: Union[IPv4Address, IPv6Address]) -> bool:
     """Check if an address is loopback or private."""
     return is_loopback(address) or is_private(address)
+
+
+def is_ip_address(address: Optional[str]) -> bool:
+    """Check if a given string is an IP address."""
+    if address is None:
+        return False
```

Comment:
I am generally not a fan of `None` checks in such functions, because you need to do a `None` check somewhere anyway, so better do it before you call this function.

Reply:
The alternative is casting it all to str() in here, which is safe as well.

Doing it before the function will riddle methods like `async_get_url` with either str casts or none checks 🤷  (caused by yarl having the host part defined as optional) (originally I had you suggested approach).

I'm fine with it either way...
Remove the None check? Or keep it?

Reply:
can we make a wrapper function that does the None check and then calls the util function, and keep that function private ?